src/AutoRest.CSharp/DataPlane/AutoRest/DataPlaneOutputLibrary.cs

```diff
@@ -207,17 +211,70 @@ public DataPlaneRestClient FindRestClient(OperationGroup operationGroup)
             return restClients;
         }
 
-        private static IEnumerable<LowLevelClientMethod> GetProtocolMethods(OperationGroup operationGroup, RestClientBuilder restClientBuilder, BuildContext<DataPlaneOutputLibrary> context)
+        // Get a Dictionary<operationGroupName, List<methodNames>> based on the "protocol-method-list" config
+        private Dictionary<string, List<string>> GetProtocolMethodsDictionary()
         {
-            // At least one protocol methods is found in the config
-            if (operationGroup.Operations.Where(IsProtocolMethodExists).Any())
+            Dictionary<string, List<string>> protocolMethodsDictionary = new();
+            foreach (var operationId in Configuration.ProtocolMethodList)
+            {
+                var operationGroupKeyAndIdArr = operationId.Split('_');
+
+                // If "operationGroup_operationId" passed in the config
+                if (operationGroupKeyAndIdArr.Length > 1)
+                {
+                    var operationGroupKey = operationGroupKeyAndIdArr[0];
+                    var methodName = operationGroupKeyAndIdArr[1];
+                    AddToProtocolMethodsDictionary(protocolMethodsDictionary, operationGroupKey, methodName);
+                }
+                // If operationGroup is not present, only operationId is passed in the config
+                else
+                {
+                    // when operationGroup is not present use the default name which is used to create client name
+                    var defaultClientName = ClientBuilder.GetClientPrefix(string.Empty, _context);
+                    AddToProtocolMethodsDictionary(protocolMethodsDictionary, defaultClientName, operationId);
+                }
+            }
+
+            return protocolMethodsDictionary;
+        }
+
+        private void AddToProtocolMethodsDictionary(Dictionary<string, List<string>> protocolMethodsDictionary, string operationGroupKey, string methodName)
```

Comment:
Can this be static?

Reply:
Using the private `_context` in [here](https://github.com/Azure/autorest.csharp/blob/66398eee793fc828588cd56079af2c42abcb5704/src/AutoRest.CSharp/DataPlane/AutoRest/DataPlaneOutputLibrary.cs#L233).

Reply:
But that is a different method. Also, `ClientBuilder.GetClientPrefix(string.Empty, _context)` can be acquired once for the whole method.

Reply:
Updated! 

Used static for `AddToProtocolMethodsDictionary` method and made `_defaultClientName` private class field.